src/tests/nodes/common_nodes.hpp

```diff
@@ -17,15 +17,74 @@
 
 #pragma once
 
+#include "srf/data/reusable_pool.hpp"
+#include "srf/memory/literals.hpp"
+#include "srf/node/generic_source.hpp"
 #include "srf/node/rx_sink.hpp"
 #include "srf/node/rx_source.hpp"
 
 namespace test::nodes {
 
+using namespace srf::memory::literals;
+
+template <std::size_t Bytes>
+class Buffer
+{
+  public:
+    Buffer() = default;
+
+    DELETE_COPYABILITY(Buffer);
+
+    void* data()
+    {
+        return m_buffer.data();
+    }
+
+    const void* data() const
+    {
+        return m_buffer.data();
+    }
+
+    std::size_t size() const
+    {
+        return m_buffer.size();
+    }
+
+  private:
+    std::array<char, Bytes> m_buffer;
+};
+
+template <std::size_t Bytes>
+class BufferSource final : public srf::node::GenericSource<srf::data::Reusable<Buffer<Bytes>>>
```

Comment:
Somewhat of a tangent, but I wasn't aware we still had GenericNodes. Unless there is some kind of functionality that can't be achieved using an RxNode, I'm kind of opposed to creating/using multiple core node types.

Reply:
Is there some specific need to support conventional inheritance for RxNodes? This seems like something a 3rd party consumer could do if they really wanted to; but kind of redundant/confusing in the base library.

Reply:
It's a convenience class for those that do not know or want to know rx.

Reply:
Off-topic - Marking as resolved.